packages/smart-wallet/src/smartWallet.js

```diff
@@ -248,27 +259,46 @@ const behavior = {
       });
 
       updatePublishKit.publisher.publish({ updated: 'brand', descriptor });
+
+      // deposit queued payments
+      const payments = paymentQueues.has(desc.brand)
+        ? paymentQueues.get(desc.brand)
+        : [];
+      const deposits = payments.map(p =>
+        // @ts-expect-error deposit does take a FarRef<Payment>
+        E(purse).deposit(p),
+      );
+      Promise.all(deposits).catch(err =>
+        console.error('ERROR depositing queued payments', err),
+      );
     },
   },
   /**
    * Similar to {DepositFacet} but async because it has to look up the purse.
    */
-  // TODO(PS0) decide whether to match canonical `DepositFacet'. it would have to take a local Payment.
+  // TODO(PS0) decide whether to match canonical `DepositFacet'. it would have to take a local Payment
   deposit: {
     /**
-     * Put the assets from the payment into the appropriate purse
+     * Put the assets from the payment into the appropriate purse.
+     *
+     * If the purse doesn't exist, we hold the payment until it does.
      *
      * @param {import('@endo/far').FarRef<Payment>} payment
-     * @returns {Promise<Amount>}
-     * @throws if the purse doesn't exist
-     * NB: the previous smart wallet contract would try again each time there's a new issuer.
-     * This version does not: 1) for expedience, 2: to avoid resource exhaustion vulnerability.
+     * @returns {Promise<Amount>} amounts for deferred deposits will be empty
      */
     async receive(payment) {
       /** @type {State} */
-      const { brandPurses } = this.state;
+      const { brandPurses, paymentQueues: queues } = this.state;
       const brand = await E(payment).getAllegedBrand();
       const purse = brandPurses.get(brand);
```

Comment:
This get will throw rather than returning undefined, no?

Reply:
yes good catch!